### ISSUE TITLE
Rule proposal: disallow usage of eslint-disable-line in favor of eslint-disable-next-line

### ISSUE THREAD

**Issue by aberonni** [mysticatea/eslint-plugin-eslint-comments#18]
_I'm duplicating this issue from https://github.com/eslint/eslint/issues/11247 because it was suggested that this rule could be a better fit here_

**Please describe what the rule should do:**

The rule should enforce using `eslint-disable-next-line` instead of using `eslint-disable-line`. This requirement comes about because of issues when integrating eslint with prettier, as discussed here:

https://github.com/prettier/prettier/issues/1267
https://github.com/prettier/prettier/issues/2273
https://github.com/prettier/prettier/issues/3011

Running prettier together with eslint can cause issues when prettier moves `eslint-disable-line` comments to the previous line, thus breaking its' intended behaviour. This can easily be solved by making sure that only the `eslint-disable-next-line` style is used, but there is no way for us to enforce this behaviour.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Bad code

```js
alert('foo'); // eslint-disable-line

alert('foo'); /* eslint-disable-line */

alert('foo'); // eslint-disable-line no-alert

alert('foo'); /* eslint-disable-line no-alert */
```

Good code

```js
// eslint-disable-next-line
alert('foo');

/* eslint-disable-next-line */
alert('foo');

// eslint-disable-next-line no-alert
alert('foo');

/* eslint-disable-next-line no-alert */
alert('foo');
```

**Comment by mysticatea** [comment]
Thank you for this proposal.

But I think you can configure it with [eslint-comments/no-use](https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/no-use.html) rule.
